src/common/snippets/include/snippets/runtime_configurator.hpp

```diff
@@ -96,11 +97,22 @@ class RuntimeConfigurator {
      * @param linear_ir LinearIR
      */
     virtual void init_tensor_rank(const lowered::LinearIRPtr& linear_ir) const;
+
+    struct UnifiedLoopInfoRtParams {
+        size_t work_amount = 0;
+        std::vector<int64_t> ptr_increments;
+        std::vector<int64_t> finalization_offsets;
+    };
+    static UnifiedLoopInfoRtParams compute_runtime_params(const lowered::UnifiedLoopInfoPtr& unified_loop_info);
+
+    using LoopInfoRuntimeParamsMap = std::unordered_map<lowered::UnifiedLoopInfoPtr, UnifiedLoopInfoRtParams>;
     /**
      * @brief Update Loop informations in LinearIR: Unified and ExpandedLoopInfo
      * @param linear_ir LinearIR
+     * @param initializated_info_map Reference on a map [LoopInfo->RuntimeParams].
+     * Can be used to pass in the method loop infos which were already initialized, e.g. by parallel domain optimization
      */
-    void update_loop_info(const lowered::LinearIRPtr& linear_ir) const;
+    void update_loop_info(const lowered::LinearIRPtr& linear_ir, LoopInfoRuntimeParamsMap& initializated_info_map) const;
```

Comment:
To discuss: if `ParallelWAOptimizer` was in common RuntimeConfig, `update_loop_info` signature would not be changed.

Reply:
It was decided to move `ParallelWAOptimizer` to a base class. However, `initializated_info_map` is kept as `update_loop_info` parameter in order to have all the `ParallelWAOptimizer` related logic in one place